packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -325,6 +412,141 @@ export class ModelConnectionTransformer extends Transformer {
+
+        // Check that related type exists and that the connected object is annotated with @model.
+        const relatedTypeName = getBaseType(field.type)
+        const relatedType = ctx.inputDocument.definitions.find(
+            d => d.kind === Kind.OBJECT_TYPE_DEFINITION && d.name.value === relatedTypeName
+        ) as ObjectTypeDefinitionNode | undefined
+
+        // Get Child object's table.
+        const tableLogicalID = ModelResourceIDs.ModelTableResourceID(relatedType.name.value);
+        const tableResource = ctx.getResource(tableLogicalID) as Table;
+
+        // Ensure that there is at least one field provided.
+        if (numFields === 0) {
+            throw new InvalidDirectiveError('No fields passed in to @connection directive.')
+        }
+
+        // Check that each field provided exists in the parent model and that it is a valid key type (single non-null).
+        let inputFields : FieldDefinitionNode[] = [];
+        args.fields.forEach(
+            item => {
+                const fieldsArrayLength = inputFields.length;
+                inputFields[fieldsArrayLength] = parent.fields.find(f => f.name.value === item);
+                if (!inputFields[fieldsArrayLength]) {
+                    throw new InvalidDirectiveError(item + ' is not a field in ' + parentTypeName)
+                }
+
+                validateKeyFieldNewConnection(inputFields[fieldsArrayLength]);
+            })
+
+        let index : GlobalSecondaryIndex = undefined;
+        // If no index is provided use the default index for the related model type and
+        // check that the query fields match the PK/SK of the table. Else confirm that index exists.
+        if (!args.keyName) {
+            checkFieldsAgainstIndex(parent.fields, relatedType.fields, args.fields, tableResource.Properties.KeySchema);
+        } else {
+            index = (tableResource.Properties.GlobalSecondaryIndexes ?
+                tableResource.Properties.GlobalSecondaryIndexes.find(GSI => GSI.IndexName === args.keyName) : null)
+                || (tableResource.Properties.LocalSecondaryIndexes ?
+                tableResource.Properties.LocalSecondaryIndexes.find(LSI => LSI.IndexName === args.keyName) : null)
+            if (!index) {
+                throw new InvalidDirectiveError('Key ' + args.keyName + ' does not exist for model ' + relatedTypeName)
+            }
+
+            // Confirm that types of query fields match types of PK/SK of the index being queried.
+            checkFieldsAgainstIndex(parent.fields, relatedType.fields, args.fields, <KeySchema[]>index.KeySchema);
+        }
+
+        // If the related type is not a list, the index has to be the default index and the fields provided must match the PK/SK of the index.
+        if (!isListType(field.type)) {
+            if (args.keyName) {
+                throw new InvalidDirectiveError('Connection is to a single object but the keyName provided does not reference the default table.')
```

Comment:
Add the ```keyName``` to the error message to make it easier for the developer to locate the error.